### PR TITLE
Configuration values for starting HSV and speed

### PR DESCRIPTION
**Pull request by robhaswell**
## Description

This PR allows the user to define starting values for the RGB matrix configuration similar to the configuration value `RGB_MATRIX_STARTUP_MODE`.

The use-case for this change is for owners of devices without the ability to write to the eeprom or otherwise permanently store the keyboard state, for example Massdrop keyboards. I am the owner of a Massdrop ALT and am using this to configure my default RGB.

It introduces the following configuration values for `config.h`:

* `RGB_MATRIX_STARTUP_HUE`
* `RGB_MATRIX_STARTUP_SAT`
* `RGB_MATRIX_STARTUP_VAL`
* `RGB_MATRIX_STARTUP_SPD`

They have the same ranges as `rgb_matrix_config.hsv`.

This PR does not include any documentation, but if this change is includable I will happily add documentation.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [x] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

Unknown.

## Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by drashna** [comment]
In theory, you can do this by saving the settings in `eeconfig_init_*`, as well.

**Comment by robhaswell** [comment]
Re `eeconfig_init_*`, yeah I did wonder today abut how this might interact with values saved in the eeprom. I don't have a board with a compatible eeprom so I can't really test it out.

On the one hand, the eeprom should be usable. On the other hand, I guess someone might want to use this even if they have an eeprom - say, they always wanted the board to start in a known configuration, for example a demo unit.

I think if these values are defined they should override the eeprom values. That way the user has the option of relying on their eeprom by not using this feature, yet could define the values to override their eeprom.

**Comment by drashna** [comment]
Well, the way it's set up in your PR, it amounts to the same thing. So it's all good.

**Comment by robhaswell** [comment]
Is this awaiting a review because it needs documentation? My experience with open source projects in the past has lead me to avoid extra effort until a change has been approved conceptually. If you think this feature is useful I'll add documentation.

**Comment by drashna** [comment]
Yeah, we'd like to have the settings documented, at least.  

Otherwise, the issue is that documentation may not get added.

**Comment by robhaswell** [comment]
I have added documentation for this now.

**Comment by robhaswell** [comment]
Good spot, I have added documentation for `RGB_MATRIX_STARTUP_SPD`.

EEPROM interaction - I think my preference (as discussed earlier) is to have these override the EEPROM at startup if defined. However, I don't instinctively know what change would be needed to do that and I also don't have an EEPROM-enabled board to test it against. With respect to intended effects, I don't have any intention in this area. I suppose if it is acceptable in its current state I think it is mergeable. "Don't let 'great' be the enemy of 'good'."

`_STARTUP_` - I agree, unfortunately that ship has sailed, we can't change the existing name and I think inconsistency is the greater evil. This is why "naming things" is one of the two greatest challenges in programming (along with "cache invalidation" and "off-by-one errors").

File: docs/feature_rgb_matrix.md [hunk 1]
```diff
@@ -380,6 +380,10 @@ These are defined in [`rgblight_list.h`](https://github.com/qmk/qmk_firmware/blo
 #define RGB_MATRIX_LED_FLUSH_LIMIT 16 // limits in milliseconds how frequently an animation will update the LEDs. 16 (16ms) is equivalent to limiting to 60fps (increases keyboard responsiveness)
 #define RGB_MATRIX_MAXIMUM_BRIGHTNESS 200 // limits maximum brightness of LEDs to 200 out of 255. If not defined maximum brightness is set to 255
 #define RGB_MATRIX_STARTUP_MODE RGB_MATRIX_CYCLE_LEFT_RIGHT // Sets the default mode, if none has been set
+#define RGB_MATRIX_STARTUP_HUE 0 // Sets the default hue value, if none has been set
+#define RGB_MATRIX_STARTUP_SAT 255 // Sets the default saturation value, if none has been set
+#define RGB_MATRIX_STARTUP_SPD 127 // Sets the default animation speed, if none has been set
+#define RGB_MATRIX_STARTUP_VAL RGB_MATRIX_MAXIMUM_BRIGHTNESS // Sets the default brightness value, if none has been set
```
Comment (fauxpark) [inline docs/feature_rgb_matrix.md]: Swap these to preserve the ordering.

Comment (robhaswell) [reply docs/feature_rgb_matrix.md]: Jesus, what is wrong with me.

**Comment by zvecr** [comment]
Thanks!

**Pull request merged**